### PR TITLE
Bug 1907874: Gherkin Script Scenarios for epic ODC-5057

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4839

Problem: Gherkin Scripts for scenarios for [ODC-5057](https://issues.redhat.com/browse/ODC-5057)

Solution: These are the gherkin scripts designed covering all the scenarios related to epics [ODC-5057](https://issues.redhat.com/browse/ODC-5057)

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 1]
```diff
@@ -55,3 +55,46 @@ Feature: Developer Catalog Page
...
+        And user clicks on Reload button when asked to
+        And user goes to Developer Catalog page
+        Then user will not see Catagories
+
+
+    @smoke, @manual
+    Scenario: Developer Catalog - Catagories under Schema tab
+        Given user is at cluster YAML tab
+        When user clicks on View sidebar
+        And user clicks on View Details under spec
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:80]:
```suggestion
        When user clicks on View Details under spec in sidebar
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 2]
```diff
@@ -55,3 +55,46 @@ Feature: Developer Catalog Page
...
+        And user clicks on customization
+        And user clicks on View Details under developerCatalog
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:82]:
```suggestion
        And user clicks on View Details under customization section in sidebar
        And user clicks on View Details under developerCatalog section in sidebar
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 3]
```diff
@@ -55,3 +55,46 @@ Feature: Developer Catalog Page
...
+        And user clicks on View Details under catagories
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:83]:
```suggestion
        And user clicks on View Details under categories
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 4]
```diff
@@ -55,3 +55,46 @@ Feature: Developer Catalog Page
...
+        Given user is at cluster YAML tab
+        And user has removed all the Catagories from Developer Catalog page
+        And user has entered "[]" in front of spec.customization.developerCatalog.catagories
+        When user clicks on View sidebar
+        And user clicks on Snippets
+        And user removes "[]" in front of spec.customization.developerCatalog.catagories
+        And user clicks on Insert Snippet
+        And user removes Languages Catagory
+        And user clicks on Save button
+        And user clicks on Reload button when asked to
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:98]:
```suggestion
        And user clicks on Reload button
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 5]
```diff
@@ -55,3 +55,46 @@ Feature: Developer Catalog Page
...
+        Then user will see all the catagories except Languages added under spec.customization.developerCatalog.catagories
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:99]:
```suggestion
        Then user will see all the categories except Languages added under spec.customization.developerCatalog.catagories
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 6]
```diff
@@ -55,3 +55,46 @@ Feature: Developer Catalog Page
...
+        And user will see all the catagories except Languages on Developer Catalog page
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:100]:
```suggestion
        And user will see all the categories except Languages on Developer Catalog page
```

**Review by makambalaji — CHANGES_REQUESTED** [review]
Please add @addFlow for all scenarios

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 7]
```diff
@@ -55,3 +55,45 @@ Feature: Developer Catalog Page
...
+        When user clicks on View sidebar
+        And user clicks on View Details under spec
+        And user clicks on customization
+        And user clicks on View Details under developerCatalog
+        And user clicks on View Details under catagories
+        Then user will see categories which are shown in the developer catalog
+
+
+    @smoke, @manual
+    Scenario: Developer Catalog Customization - Edit Catagories
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 8]
```diff
@@ -55,3 +55,45 @@ Feature: Developer Catalog Page
...
+        And user removes everything from spec.customization.developerCatalog.catagories
+        And user enters "[]" in front of spec.customization.developerCatalog.catagories
+        And user clicks on Save button
+        And user clicks on Reload button when asked to
+        And user goes to Developer Catalog page
+        Then user will not see Catagories
+
+
+    @smoke, @manual
+    Scenario: Developer Catalog Customization - Catagories
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 9]
```diff
@@ -55,3 +55,45 @@ Feature: Developer Catalog Page
...
+        And user searches for Console
+        And user clicks on Console checkbox with "operator.openshift.io/v1"
+        And user clicks on cluster link
+        And user goes to YAML tab
+        And user removes everything from spec.customization.developerCatalog.catagories
+        And user enters "[]" in front of spec.customization.developerCatalog.catagories
+        And user clicks on Save button
+        And user clicks on Reload button when asked to
+        And user goes to Developer Catalog page
+        Then user will not see Catagories
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 10]
```diff
@@ -55,3 +55,45 @@ Feature: Developer Catalog Page
         And user will see the cards of Helm Charts
         And user will see Filter by Keyword field
         And user will see A-Z, Z-A sort by dropdown
+
+
+    @smoke, @manual
+    Scenario: Developer Catalog Customization - Add empty Catagories
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 11]
```diff
@@ -55,3 +55,45 @@ Feature: Developer Catalog Page
...
+        Given user is at Search page
+        When user clicks on Resources dropdown
+        And user searches for Console
+        And user clicks on Console checkbox with "operator.openshift.io/v1"
+        And user clicks on cluster link
+        And user goes to YAML tab
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 12]
```diff
@@ -55,3 +55,45 @@ Feature: Developer Catalog Page
...
+        And user removes everything from spec.customization.developerCatalog.catagories
+        And user enters "[]" in front of spec.customization.developerCatalog.catagories
+        And user clicks on Save button
+        And user clicks on Reload button when asked to
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 5 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated

**Comment by jerolimov** [comment]
looks like it tests all required cases :+1:

/lgtm

**Pull request merged**